geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/KryoFeatureSerializer.scala

```diff
@@ -15,6 +15,27 @@ import org.locationtech.geomesa.features.kryo.impl.LazyDeserialization.{Immutabl
 import org.locationtech.geomesa.features.kryo.impl.{KryoFeatureDeserialization, KryoFeatureSerialization}
 import org.opengis.feature.simple.SimpleFeatureType
 
+/**
+  * Kryo feature serialization and deserialization.
+  *
+  * The current serialization scheme (version 3):
+  *
+  * <ol>
+  *   <li>one byte containing the serialization version (3)</li>
+  *   <li>two bytes for a short containing the number of serialized attributes. this supports
+  *       appending attributes to the schema, as we know how many attributes were written when deserializing</li>
+  *   <li>a known number of bytes for metadata, which consists of:
```

Comment:
Do we have unit tests which verify that one can add attributes and read 'old' data?

Reply:
yeah, we have the 'back compatibility tests' and there's also an accumulo test here: https://github.com/locationtech/geomesa/blob/master/geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreAlterSchemaTest.scala